### DISCUSSION TITLE
방명록 순서 및 다크/라이트 모드에 따른 글자 색상 설정, 댓글 설정 관련

### DISCUSSION THREAD

**Discussion by eehoeskrap** [pronist/hELLO#112] [Ideas]
안녕하세요. 일단 훌륭한 스킨 만들어주셔서 감사합니다. 
오랜 시간 동안 티스토리 블로그를 운영하면서 기본 스킨만 써오다가 hELLO 스킨이 너무 깔끔해보여서 적용을 했는데요!
몇 가지 궁금한 사항이 있어서 discussion 남기게 되었습니다. 

**1. 방명록 순서**
방명록에 들어가면 맨 상단에 최근 방명록이 표시된게 아니라 해당 페이지의 제일 오래된 방명록이 뜹니다. 
예를 들면 아래와 같습니다. (사진을 첨부하고 싶은데 방명록 내용이 다 보이는 관계로 텍스트로 적습니다.)

페이지 1
- 2020.11.17
- 2020.12.17
- 2022.06.25

페이지 2
- 2016.09.21 
- 2017.09.14
- 2020.05.13

**2.  다크/라이트 모드에 따른 글자 색상 설정**

저 같은 경우 오랜 시간동안 라이트 모드에 기반한 스킨을 써왔고,
글 작성 할 때 글자 색상이 회색 계열로 되어있어서 항상 검은색으로 임의로 지정하고 사용을 했었는데요.
그러다 보니 hELLO 스킨의 다크모드를 적용하면 다크 배경에 검은색 글자로 표시가 됩니다 😢 
다크 모드를 적용했을 때 게시글의 글자 색상을 화이트로 일괄 변경 할 수는 없을까요?

지금 스킨이 기본 글자 색상(색상 설정하지 않은 경우)은 다크모드일때 화이트로 변경되고,
라이트 모드일 때 블랙으로 변경 되기에 
기본 글자 색상을 임의로 모드에 따라 화이트나 블랙으로 픽스 해버릴 경우 
사용자가 글자 색상을 지정 하였을 때 (예를 들면 레드) 색이 지정되지 않는 문제가 발생할 것 같긴 하지만...

혹시 모드 변경에 따른 게시글 글자 색상 변경 방법 없을까 하여 질문 드립니다. 
없다면... 테마 모드 변경 버튼을 없애고 디폴트 값을 라이트 모드로 설정하는 방법 알려주시면 감사 드리겠습니다... 😀 

**3. 댓글 보기 설정**
글에 달린 댓글이나 방명록에 댓글을 보려면 "댓글 보기" 버튼을 눌러야 보이게 되어있는데,
디폴트를 댓글이 다 보이게끔 설정해두고 댓글 접기 버튼을 생성할 수 있는 방법이 궁금합니다. 

이상입니다. 답변 기다리고 있겠습니다! 
감사합니다

**Comment by pronist** [comment]
@eehoeskrap 안녕하세요!

1. 방명록 순서

이 부분은 제가 스킨을 만들면서 미쳐 신경쓰지 못한 부분이므로 다음 버전에서 패치됩니다.

2. 다크/라이트 모드 글자 색상 설정

기존 티스토리 게시 글의 html 에서 span 태그로 글자의 색상이 변경되어 있는 경우 반영되기 위해 다크모드에서 강제로 색상변경처리를 하지 않습니다. 이 부분은 어떤 사람은 강제로 해달라고 하고, 또 누군가는 유지해주었으면 하는 그런 상황입니다. 따라서 개발자는 선택할 수 밖에 없는데 색상지정을 유지하는 방향으로 처리한 것입니다. 강제로 색상을 변경하려면 CSS 를 수정하는 방법 이외에는 없습니다.

현재 다크모드의 테마는 <https://github.com/tistory-projects/tistory-skin-hELLO/blob/master/themes/dark.css> 이며 이를 별도로 다운받아서 수정하시고 link 태그를 아래와 같이 변경하길 바랍니다. 

```html
<link rel="stylesheet" href="./images/dark.css">
```

또는 별도의 CSS 를 작성하시고 덮어씌우는 것도 방법이겠죠.

3. 댓글 펼치기

댓글 또한 누군가는 접어달라고 하고 누군가는 펼쳐달라고 합니다. 따라서 접는 것을 택했습니다. 다만 이 경우, 펴는 옵션의 제공여부는 다음 패치에서 결정될 예정입니다.

**Reply by eehoeskrap** [reply]
감사합니다. 다음 버전 기다리고 있겠습니다 :) 좋은 하루 되세요!
